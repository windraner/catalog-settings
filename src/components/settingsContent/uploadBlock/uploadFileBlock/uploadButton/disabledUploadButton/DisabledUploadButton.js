import React, { Component } from 'react';
import ToolTip from '../../../toolTip/ToolTip';

export default class DisabledUploadButton extends Component {
  state = {
    isTooltipShow: false
  }

  hoverHandler = (value) => {
    this.setState({isTooltipShow: value});
  }

  render() {
    const { isTooltipShow } = this.state;

    return (
      <div
        onMouseEnter={() => this.hoverHandler(true)}
        onMouseLeave={() => this.hoverHandler(false)}
        className="catalog-settings-content__upload-button catalog-settings-content__upload-button_disabled"
      >
        <ToolTip
          isShow={isTooltipShow}
          text={'Remove URL to manually add CSV file'}
        />
        <span className="catalog-settings-content__upload-button-icon"></span>
        <span>Upload file</span>
      </div>
    );
  }
}
