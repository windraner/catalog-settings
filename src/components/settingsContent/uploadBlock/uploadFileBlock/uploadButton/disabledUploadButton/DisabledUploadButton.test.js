import React from 'react';
import DisabledUploadButton from './DisabledUploadButton';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<DisabledUploadButton />', () => {
  const wrapper = shallow(<DisabledUploadButton />);
  it('Check for className and text', () => {
    expect(wrapper.find('.catalog-settings-content__upload-button').length).toBe(1);
    expect(wrapper.find('.catalog-settings-content__upload-button').text()).toBe('<ToolTip />Upload file');
  });
  it('Check hover change state', () => {
    wrapper.find('.catalog-settings-content__upload-button').simulate('mouseEnter');
    expect(wrapper.state('isTooltipShow')).toBe(true);
    wrapper.find('.catalog-settings-content__upload-button').simulate('mouseLeave');
    expect(wrapper.state('isTooltipShow')).toBe(false);
  });
  it('matches the snapshot', () => {
    const tree = shallow(<DisabledUploadButton />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
