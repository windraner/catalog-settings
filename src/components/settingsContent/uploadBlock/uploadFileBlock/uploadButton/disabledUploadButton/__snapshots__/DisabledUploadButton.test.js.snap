// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisabledUploadButton /> matches the snapshot 1`] = `
<div
  className="catalog-settings-content__upload-button catalog-settings-content__upload-button_disabled"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <ToolTip
    isShow={false}
    text="Remove URL to manually add CSV file"
  />
  <span
    className="catalog-settings-content__upload-button-icon"
  />
  <span>
    Upload file
  </span>
</div>
`;
